test/jdk/com/sun/management/DiagnosticCommandMBean/DcmdMBeanTestCheckJni.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021, Red Hat, Inc.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug     8258836
+ * @summary JNI local refs exceed capacity getDiagnosticCommandInfo
+ * @library /test/lib
+ * @run main/othervm DcmdMBeanTestCheckJni
+ */
+
+import java.lang.management.ManagementFactory;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.MBeanServerConnection;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorServerFactory;
+import javax.management.remote.JMXServiceURL;
+import javax.management.remote.JMXConnectorServer;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+public class DcmdMBeanTestCheckJni {
+
+    public static void main(String[] args) throws Exception {
+        OutputAnalyzer out = ProcessTools.executeTestJvm(
+            "-Xcheck:jni",
+            DcmdMBeanRunner.class.getName());
+        out.shouldNotMatch("WARNING: JNI local refs: \\d+, exceeds capacity: \\d+")
+           .shouldContain("DcmdMBeanRunner COMPLETE")
+           .shouldHaveExitValue(0);
+    }
+
+}
+
+class DcmdMBeanRunner {
+
+    private static String HOTSPOT_DIAGNOSTIC_MXBEAN_NAME =
```

Comment:
Excess newline and the filed could be `static final`. Actually, you might even go and merge both classes, and then pass a fake java arg in `main` to disambiguate the "driver" and "test" code, like:

```
    public static void main(String[] args) {
        if (args.length == 0) {
            driver();
        } else {
            runner();
        }
    }
```

Your call.

Reply:
Thanks. I've made `HOTSPOT_DIAGNOSTIC_MXBEAN_NAME` final too. The line break is so that the line doesn't exceed 80 chars. Not sure if that's still something to consider, but it shouldn't really do any harm, so I've left it. I'm on the fence about `DcmdMBeanRunner` class. Seems a tad easier to read to me as is rather than the conditional code. Hope that's OK.